pkg/apis/crd/v1alpha1/types.go

```diff
@@ -354,3 +355,137 @@ type BGPPeer struct {
+	DstPort *int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	// IPFamily is the filter's IP family. Default to `IPv4`.
+	IPFamily v1.IPFamily `json:"ipFamily,omitempty"`
+	// Protocol represents the transport protocol. Default is to not filter on protocol.
+	Protocol        *intstr.IntOrString `json:"protocol,omitempty"`
+	TransportHeader TransportHeader     `json:"transportHeader"`
+}
+
+// PacketCaptureFirstNConfig contains the config for the FirstN type capture. The only supported parameter is
+// `Number` at the moment, meaning capturing the first specified number of packets in a flow.
+type PacketCaptureFirstNConfig struct {
+	Number int32 `json:"number"`
+}
+
+const DefaultPacketCaptureTimeout uint16 = 60
+
+type PacketCapturePhase string
+
+const (
+	PacketCaptureRunning   PacketCapturePhase = "Running"
+	PacketCaptureSucceeded PacketCapturePhase = "Succeeded"
+	PacketCaptureFailed    PacketCapturePhase = "Failed"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCaptureList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PacketCapture `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCapture struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              PacketCaptureSpec   `json:"spec"`
+	Status            PacketCaptureStatus `json:"status"`
+}
+
+type CaptureConfig struct {
+	FirstN *PacketCaptureFirstNConfig `json:"firstN,omitempty"`
+}
+
+// PacketCaptureFileServer specifies the PacketCapture file server information.
+type PacketCaptureFileServer struct {
+	// The URL of the file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string `json:"url"`
+}
+
+type PacketCaptureSpec struct {
+	Timeout       *uint16       `json:"timeout,omitempty"`
+	CaptureConfig CaptureConfig `json:"captureConfig"`
+	Source        Source        `json:"source"`
+	Destination   Destination   `json:"destination"`
+	Packet        *Packet       `json:"packet,omitempty"`
+	// FileServer specifies the sftp url config for a file server. If present, captured packets will be uploaded to this server.
+	FileServer *PacketCaptureFileServer `json:"fileServer,omitempty"`
+}
+
+type PacketCaptureStatus struct {
+	Phase PacketCapturePhase `json:"phase"`
+	// Reason records the failure reason when the capture fails.
+	Reason string `json:"reason"`
+	// NumCapturedPackets records how many packets have been captured. If it reaches the target number, the capture
+	// can be considered as finished.
+	NumCapturedPackets *int32 `json:"numCapturedPackets,omitempty"`
```

Comment:
I feel this doesn't need to be a pointer. It's correct to use `NumCapturedPackets=0` to represent nothing has been captured.

Reply:
ok